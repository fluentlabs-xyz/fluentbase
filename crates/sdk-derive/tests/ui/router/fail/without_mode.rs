#![allow(dead_code)]

extern crate alloc;
extern crate fluentbase_sdk;

use fluentbase_sdk::{
    basic_entrypoint,
    derive::{router, Contract},
};

#[derive(Contract)]
struct MyContract<SDK> {
    sdk: SDK,
}

#[router]
impl<SDK: SharedAPI> MyContract<SDK> {
    fn private_method(&self, a: u32) -> u64 {
        a as u64
    }
}

basic_entrypoint!(MyContract);
