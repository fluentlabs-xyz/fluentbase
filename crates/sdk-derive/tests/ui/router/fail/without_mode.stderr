error: Missing field `mode`
  --> tests/ui/router/fail/without_mode.rs:16:1
   |
16 | #[router]
   | ^^^^^^^^^
   |
   = note: this error originates in the attribute macro `router` (in Nightly builds, run with -Z macro-backtrace for more info)
